File: functions/updatebudget.c
#include <stdio.h>
float income, rent, utilities, groceries, transportation, savings, expenses, spend;

float input(char type[], float var){
  printf("Monthly %s:\n", type);
  scanf("%f", &var);
  return var;
}

void percent(char type[], int amount){
  int per = amount/income *100;

printf("Your %s is %d%% of your income.\n", type, per);
}

int main(void){

  printf("This is your budget calculator.\n");
  income = input("income", income);
  rent = input("rent", rent);
  utilities = input("utilities", utilities);
  groceries = input("groceries", groceries);
  transportation = input("trancportation", transportation);
  savings = income * .2;
  expenses = rent + utilities + groceries + transportation;
  spend = income - expenses - savings;
  printf("You make $%.2\n", income);
  printf("Your expenses are $%.2\n", expenses);
  printf("Your savings are $%.2\n", savings);
  printf("Your spending money is $%.2\n", spend);
  percent("rent", rent);
  percent("utilities", utilities);
  percent("groceries", groceries);
  percent("transportation", transportation);
  percent("expenses", expenses);
  percent("savings", savings);
  return 0;
}

File: functions/updatebudget.py
def float(type):
    return float(input (f"what is your {type}: "))
print("Hello, and welcome to your financial calculator!\n")
income=float(input("income cost"))
rent=float(input("rent cost"))
utilities=float(input("utilities cost "))
groceries=float(input("groceries cost "))
transportation=float(input("transportation costs"))
savings = income *.2
expenses = rent+utilities+groceries+transportation
spending = income-expenses-savings
def percent(type, amount):
    per = amount/income *100

    return f"Your {type} is {per}% income."

print(f"Your monthly income is ${income:.2f}\n")
print(f"Your monthly expenses are ${expenses:.2f}\n")
print(f"Your monthly savings is ${savings:.2f}\n")
print(f"Your monthly spending money is ${spending:.2f}\n")
percent("rent", rent)
percent("utilities", utilities)
percent("groceries", groceries)
percent("transportation", transportation)
percent("savings", savings)
percent("expenses", expenses)